### PR TITLE
 [SCB-708] Aggregate simple query params into object param in spring mvc style

### PR DESCRIPTION
**Pull request by yhs0092**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
See details in [SCB-708](https://issues.apache.org/jira/browse/SCB-708)

### CODE REVIEW

File: swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/utils/ParamUtils.java [hunk 1]
```diff
@@ -27,15 +27,14 @@
...
-import com.fasterxml.jackson.databind.ser.std.MapProperty;
-
 import io.swagger.converter.ModelConverters;
 import io.swagger.models.Model;
 import io.swagger.models.Swagger;
 import io.swagger.models.parameters.AbstractSerializableParameter;
 import io.swagger.models.parameters.BodyParameter;
 import io.swagger.models.parameters.Parameter;
 import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.MapProperty;
```
Comment (wujimin) [inline swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/utils/ParamUtils.java]: import but no used?

File: swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+public class SpringmvcDefaultObjectParameterProcessor implements DefaultParameterProcessor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SpringmvcDefaultObjectParameterProcessor.class);
+
+  @Override
+  public void process(OperationGenerator operationGenerator, int paramIndex) {
+    Model paramModel = getParamModel(operationGenerator, paramIndex);
+
+    if (null == paramModel) {
+      LOGGER.warn("cannot find param, provider method is [{}], paramIndex = [{}]",
```
Comment (wujimin) [inline swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java]: just log a error and warn message?
will go with parameter lost?

File: swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    return paramRoot;
+  }
+
+  private LinkedHashMap<String, AbstractSerializableParameter<?>> getFlattenParams(Model paramModel) {
+    LinkedHashMap<String, AbstractSerializableParameter<?>> flattenParamMap = new LinkedHashMap<>();
+    // traversal the properties of current paramModel
+    // create simple parameters, nesting object param is ignored
+    for (Entry<String, Property> propertyEntry : paramModel.getProperties().entrySet()) {
+      if (ParamUtils.isComplexProperty(propertyEntry.getValue())) {
+        LOGGER.info("A nesting complex field is ignored, field name  = [{}]", propertyEntry.getKey());
```
Comment (wujimin) [inline swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java]: if need to be ignore by swagger, better to annotate by "@JsonIgnore"
if we just log and go on, maybe will cause unnotice problem.

File: swagger/swagger-invocation/invocation-core/pom.xml [hunk 4]
```diff
@@ -47,5 +47,9 @@
       <artifactId>log4j</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (wujimin) [inline swagger/swagger-invocation/invocation-core/pom.xml]: why need depend spring-web in invocation-core?

File: swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/utils/ParamUtils.java (hunk 1 above)
Comment (yhs0092) [reply swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/utils/ParamUtils.java]: In line 130，MapProperty is used for param type check. A wrong MapProperty is imported before.

Comment (wujimin) [reply swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/utils/ParamUtils.java]: ok

File: swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java (hunk 2 above)
Comment (yhs0092) [reply swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java]: OK, then should I throw an Exception to break this process?

Comment (yhs0092) [reply swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java]: Fixed, now an error will be thrown to interrupt boot up procedure

File: swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java (hunk 3 above)
Comment (yhs0092) [reply swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java]: Fixed, now an error will be thrown to interrupt boot up procedure

**Pull request merged**